sdk/go/common/apitype/search.go

```diff
@@ -0,0 +1,120 @@
+	Custom       *bool                        `json:"custom,omitempty" csv:"custom"`
+	Delete       *bool                        `json:"delete,omitempty" csv:"delete"`
+	Dependencies []string                     `json:"dependencies,omitempty" csv:"-"`
+	ID           *string                      `json:"id,omitempty" csv:"id"`
+	Matches      map[string][]json.RawMessage `json:"matches,omitempty" csv:"-"`
+	Modified     *string                      `json:"modified,omitempty" csv:"modified"`
+	Module       *string                      `json:"module" csv:"module"`
+	Name         *string                      `json:"name,omitempty" csv:"name"`
+	Package      *string                      `json:"package" csv:"package"`
+	Parent       *string                      `json:"parent.urn,omitempty" csv:"parent_urn"`
+	Pending      *string                      `json:"pending,omitempty" csv:"pending"`
+	Program      *string                      `json:"project,omitempty" csv:"project"`
+	Protected    *bool                        `json:"protected,omitempty" csv:"protected"`
+	ProviderURN  *string                      `json:"provider.urn,omitempty" csv:"provider_urn"`
+	Stack        *string                      `json:"stack,omitempty" csv:"stack"`
+	Type         *string                      `json:"type,omitempty" csv:"type"`
+	URN          *string                      `json:"urn,omitempty" csv:"urn"`
+	Teams        teamNameSlice                `json:"teams,omitempty" csv:"teams"`
+	Properties   *RawProperty                 `json:"properties,omitempty" csv:"properties"`
+}
+
+// Aggregation collects the top 5 aggregated values for the requested dimension.
+type Aggregation struct {
+	Others  *int64              `json:"others,omitempty"`
+	Results []AggregationBucket `json:"results,omitempty"`
+}
+
+// AggregationBucket is an example value for the requested aggregation, with a
+// count of how many resources share that value.
+type AggregationBucket struct {
+	Name  *string `json:"name,omitempty"`
+	Count *int64  `json:"count,omitempty"`
+}
+
+// ResourceSearchPagination provides links for paging through results.
+type ResourceSearchPagination struct {
+	Previous *string `json:"previous,omitempty"`
+	Next     *string `json:"next,omitempty"`
+	Cursor   *string `json:"cursor,omitempty"`
+}
+
+type teamNameSlice []string
+
+func (teams *teamNameSlice) MarshalCSV() (string, error) {
+	if teams == nil || len(*teams) == 0 {
+		return "", nil
+	}
+
+	json, err := json.Marshal(*teams)
+	if err != nil {
+		return "", err
+	}
+
+	return string(json), nil
+}
+
+// RawProperty wraps a json.RawProperty for JSON or CSV export.
+type RawProperty struct {
+	json.RawMessage
+}
+
+type PulumiQueryResponse struct {
+	Query string `json:"query"`
+}
+
+type PulumiQueryRequest struct {
+	Query string `url:"query"`
+}
+
+func ParseQueryParams(rawParams *[]string) *PulumiQueryRequest {
+	queryString := ""
+	for _, param := range *rawParams {
+		paramElements := strings.Split(param, "=")
```

Comment:
Should this be a `SplitN`?
What happens when the value has an `=` symbol?

Better, this can be [strings.Cut](https://pkg.go.dev/strings#Cut):

```go
k, v, ok := strings.Cut(param, "=")
```

It'll split on the first instance of `=` in the string, and will return `ok = false` if the string doesn't have an `=`.